vault/client.go

```diff
@@ -28,30 +28,51 @@ type AuthStrategy interface {
 	Revokable() bool
 }
 
+// GetenvFunc returns the supplied environment variable value
+type GetenvFunc func(string) string
```

Comment:
Maybe I just need to get over my aversion to Setenv in tests. I see os_test.go itself does it:

https://github.com/golang/go/blob/master/src/os/os_test.go#L1118